packages/react-native-reanimated/src/hook/useAnimatedStyle.ts

```diff
@@ -150,7 +151,16 @@ function runAnimations(
         animation.callback && animation.callback(true /* finished */);
       }
     }
-    result[key] = animation.current;
+    /*
+     * If `animation.current` is an object, spread its properties into a new object
+     * to avoid modifying the original reference. This ensures when `newValues` has a nested color prop, it stays unparsed
+     * in rgba format, allowing the animation to run correctly.
+     */
+    if (typeof animation.current === 'object') {
+      result[key] = { ...animation.current };
```

Comment:
Do we also need to copy other objects from the style, such as transforms? 🤔

Reply:
In this approach we also copy transform yes, but it doesn't affect it - it only makes sure any nested the color property will work